core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -725,10 +725,15 @@ private[spark] class ExecutorAllocationManager(
         if (stageIdToNumRunningTask.contains(stageId)) {
           stageIdToNumRunningTask(stageId) += 1
         }
-        // This guards against the race condition in which the `SparkListenerTaskStart`
-        // event is posted before the `SparkListenerBlockManagerAdded` event, which is
-        // possible because these events are posted in different threads. (see SPARK-4951)
-        if (!allocationManager.executorIds.contains(executorId)) {
+        // This guards against the following race condition:
+        // 1. The `SparkListenerTaskStart` event is posted before the
+        // `SparkListenerExecutorAdded` event
+        // 2. The `SparkListenerExecutorRemoved` event is posted before the
+        // `SparkListenerTaskStart` event
+        // Above cases are possible because these events are posted in different threads.
+        // (see SPARK-4951 SPARK-26927)
+        if (!allocationManager.executorIds.contains(executorId) &&
+          client.getExecutorIds().contains(executorId)) {
```

Comment:
nit: indent more

Reply:
@vanzin Seems 2 spaces indent is suggested, isn't it? I checked several scala classes of spark project, the indention of similar codes are indented by 2 spaces. What's the prefer one we suggested?

https://github.com/apache/spark/blob/57ae251f75cef4ad5478c42709870a1666ddad46/core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala#L499

Reply:
Ok, let's stop counting spaces. The goal is not 2 spaces or 4 spaces. The goal is to separate different blocks of code.

If you indent two spaces here, your condition is at the same indent as the block of code inside the `if`. So indent it more so it's clear that it's not.

Reply:
Yes, it's clear and I'm not bother on this, but I just want to make sure the 4 space indention is not breaking any rule. I will update as your suggestion. Thank you @vanzin